packages/mask/src/components/shared/ApplicationSettingPluginSwitch.tsx

```diff
@@ -0,0 +1,120 @@
+import { useActivatedPluginsSNSAdaptor, Plugin, PluginI18NFieldRender } from '@masknet/plugin-infra/content-script'
+import { SettingSwitch } from '@masknet/shared'
+import { makeStyles, MaskColorVar } from '@masknet/theme'
+import { Services } from '../../extension/service'
+interface Props {}
+const useStyles = makeStyles()((theme) => ({
+    listItem: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+        justifyContent: 'space-between',
+        borderRadius: 12,
+        '&:hover': {
+            background: theme.palette.background.default,
+        },
+        '&:hover .MuiAvatar-root': {
+            background: theme.palette.background.paper,
+        },
+    },
+    listContent: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+    },
+    headerWrapper: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+    },
+    settings: {
+        alignSelf: 'flex-start',
+        paddingTop: theme.spacing(0.5),
+        marginLeft: theme.spacing(0.5),
+        cursor: 'pointer',
+        color: MaskColorVar.textSecondary,
+        opacity: theme.palette.mode === 'dark' ? 0.5 : 1,
+    },
+    avatar: {
+        background: theme.palette.background.default,
+        width: '44px',
+        height: '44px',
+        '> *': {
+            width: 26,
+            height: 26,
+        },
+    },
+    info: {
+        maxWidth: 420,
+    },
+    name: {
+        fontSize: 14,
+        fontWeight: 700,
+    },
+    desc: {
+        fontSize: 12,
+        fontWeight: 400,
+        color: theme.palette.mode === 'dark' ? theme.palette.text.secondary : theme.palette.text.primary,
+    },
+}))
+export function ApplicationSettingPluginSwitch(props: Props) {
+    const { classes } = useStyles()
+    const snsAdaptorPlugins = useActivatedPluginsSNSAdaptor('any')
+    const snsAdaptorMinimalPlugins = useActivatedPluginsSNSAdaptor(true)
+
+    async function onSwitch(id: string, checked: boolean) {
+        await Services.Settings.setPluginMinimalModeEnabled(id, !checked)
+    }
+
+    return (
+        <List>
+            {snsAdaptorPlugins
+                .reduce<{ entry: Plugin.SNSAdaptor.ApplicationEntry; pluginId: string }[]>((acc, cur) => {
+                    if (!cur.ApplicationEntries) return acc
+                    return acc.concat(
+                        cur.ApplicationEntries.map((x) => {
+                            return {
+                                entry: x,
+                                pluginId: cur.ID,
+                            }
+                        }) ?? [],
+                    )
+                }, [])
+                .filter((x) => x.entry.category === 'dapp')
+                .sort((a, b) => (a.entry.marketListSortingPriority ?? 0) - (b.entry.marketListSortingPriority ?? 0))
+                .map((x, i) => (
+                    <ListItem key={x.pluginId + i} className={classes.listItem}>
```

Comment:
Why does plugin id still not enough for as `key`?

```suggestion
                    <ListItem key={x.pluginId} className={classes.listItem}>
```

Reply:
Jack's suggestion.